packages/typespec-ts/src/utils/credentialUtils.ts

```diff
@@ -39,3 +40,44 @@ export function getSupportedHttpAuth(
 
   return authList;
 }
+
+export function isSupportedKeyCredential(auth: HttpAuth): boolean {
+  return (
+    (auth.type === "apiKey" && auth.in === "header") || auth.type === "http"
+  );
+}
+
+export function isSupportedTokenCredential(auth: HttpAuth): boolean {
+  return auth.type === "oauth2" || auth.type === "openIdConnect";
```

Comment:
for oauth2 i think yes, this is default auth for ARM. for openIdConnect i am not sure, i copy this from jose's pr. i enable this mainly because it is kinda of [oauth2](https://www.microsoft.com/en-us/security/business/security-101/what-is-openid-connect-oidc). but i am okay to remove if we have no confidance.

Reply:
let's remove this for this pr.